core/pom.xml

```diff
@@ -517,7 +511,12 @@ THE SOFTWARE.
     <dependency>
       <groupId>com.github.spotbugs</groupId>
       <artifactId>spotbugs-annotations</artifactId>
-      <optional>true</optional>
+      <exclusions>
+        <exclusion>
+          <groupId>com.google.code.findbugs</groupId>
+          <artifactId>jsr305</artifactId>
+        </exclusion>
+      </exclusions>
```

Comment:
`spotbugs-annotations` _does_ depend on JSR 305, so we need this exclusion to keep the WAR contains identical to the contents before this change (i.e., without JSR 305). Note that we are also removing `<optional>true</optional>`. This previously had no effect because Stapler directly depended on `spotbugs-annotations` (and thus we bundled `spotbugs-annotations` into the WAR), but now that Stapler's dependency on `spotbugs-annotations` is optional, we would lose the `spotbugs-annotations` from the WAR unless we removed the optional declaration here. Keeping `spotbugs-annotations` in the WAR was the agreed-upon consensus in the mailing list thread.